# Importing Libraries

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# Importing Datasets

In [8]:
dataset = pd.read_csv('/Users/user/Desktop/data/data (1).csv')
x = dataset.iloc[ : , 2:-1]
y = dataset.iloc[ : ,-1]

# Printing x(Independent Variables) and y(Dependent Variables)

In [11]:
print(x)

    Volume  Weight
0     1000     790
1     1200    1160
2     1000     929
3      900     865
4     1500    1140
5     1000     929
6     1400    1109
7     1500    1365
8     1500    1112
9     1600    1150
10    1100     980
11    1300     990
12    1000    1112
13    1600    1252
14    1600    1326
15    1600    1330
16    1600    1365
17    2200    1280
18    1600    1119
19    2000    1328
20    1600    1584
21    2000    1428
22    2100    1365
23    1600    1415
24    2000    1415
25    1500    1465
26    2000    1490
27    2000    1725
28    1600    1523
29    2000    1705
30    2100    1605
31    2000    1746
32    1600    1235
33    1600    1390
34    1600    1405
35    2500    1395


In [13]:
print(y)

0      99
1      95
2      95
3      90
4     105
5     105
6      90
7      92
8      98
9      99
10     99
11    101
12     99
13     94
14     97
15     97
16     99
17    104
18    104
19    105
20     94
21     99
22     99
23     99
24     99
25    102
26    104
27    114
28    109
29    114
30    115
31    117
32    104
33    108
34    109
35    120
Name: CO2, dtype: int64


# Splitting Data Into Training And Test Set

In [16]:
#importing train_test_split function to help us split data
from sklearn.model_selection import train_test_split
#splitting our data into 80% training and 20% test data 
x_train, x_test, y_train, y_test = train_test_split(x,y,test_size = 0.2, random_state = 0)

# Training A Multiple Linear Regression Model On Training Set

In [19]:
from sklearn.linear_model import LinearRegression #importing linear regression model to help us not build from scratch using y=mx + c
regressor = LinearRegression() #creating an object of linearRegression class hence building the model
regressor.fit(x_train,y_train) #attaching our training set to our object or model so it is trained

LinearRegression()

# Predicting Test Set

In [22]:
y_pred = regressor.predict(x_test) #we are trying to predict the results of the test feature so we can later on compare with the actual 
# to evaluate our model
np.set_printoptions(precision=2)
print(np.concatenate((y_pred.reshape(len(y_pred),1), y_test.values.reshape(len(y_test),1)),1)) #to print the actual value and predicted value vertically

[[108.55 117.  ]
 [104.32  94.  ]
 [102.72  99.  ]
 [108.28 115.  ]
 [106.53  99.  ]
 [102.47  97.  ]
 [ 96.1   99.  ]
 [ 94.97  95.  ]]


# Making a single prediction 
(for example the profit of a startup with R&D Spend = 160000, Administration Spend = 130000, Marketing Spend = 300000 and State = 'California')

In [24]:
print(regressor.predict([[2300, 1300]]))

[107.59]


/Users/anaconda3/lib/python3.12/site-packages/sklearn/base.py:493: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


# Getting the final linear regression equation with the values of the coefficients

In [120]:
print(regressor.coef_)
print(regressor.intercept_)

[ 8.66e+01 -8.73e+02  7.86e+02  7.73e-01  3.29e-02  3.66e-02]
42467.5292485338


Therefore, the equation of our multiple linear regression model is:

Profit=86.6×Dummy State 1−873×Dummy State 2+786×Dummy State 3+0.773×R&D Spend+0.0329×Administration+0.0366×Marketing Spend+42467.53

Important Note: To get these coefficients we called the "coef_" and "intercept_" attributes from our regressor object. Attributes in Python are different than methods and usually return a simple value or an array of values.